CONTRIBUTING.md

```diff
@@ -0,0 +1,132 @@
+# Contributing to the documentation
+
+## Repository structure
+
+Briefly, this is what you need to know about the documentation repository structure before contributing:
+
+* The documentation is module-based, which means most sections are individual modules. These modules are available in the `/docs/topics/modules/` folder.
+* The modules are linked together to form a chapter, document, or guide using assemblies. These assemblies are asciidoc files in the `/docs/topics/` folder. For example, the `/docs/topics/getting-started-guide.adoc` file is the main assembly for the Getting Started Guide and links to other, smaller assemblies for parts or chapters within the guide.
```

Comment:
(Copying over from my earlier comment, once I fixed the asterix to backticks it just collapsed all my suggestions in this section.)
Suggestion: The Repository structure section could go in the Readme. From an external contributor pov I would want to know how the repo is structured, to understand the lay upfront. wdyt?
Also we could add the modular docs link here: https://redhat-documentation.github.io/modular-docs/#introduction

Reply:
+1 to adding the mod docs link as more reference material for those who are interested but not sure the repo structure is relevant unless you want to contribute to the repo. Thoughts?

Reply:
Well, imo, any user who comes to OSIO docs repository would want to know more about the Docs repository. OSIO description in short, ok (maybe a link), but definitely more about how we use this repository to create 2 guides and so on etc.

Reply:
+1 about the repo structure... The thing is... I adopted the repo structure because it was used by the RHOAR docs, and it turns out the structure wasn't exactly thought out well. So, I'm seriously considering fixing it (at the cost of breaking compatibility with RHOAR).

Reply:
> Okay, so was that a +1 from you about moving the structure to the README instead?

Yes.

> Also, can we have a separate discussion somewhere about the structure if that's changing again? I kind of like the idea of modules in a separate folder (easy to find, easy list to look through to find the correct module) and images in a separate folder, plus all assemblies in a different, higher level folder.

I don't want to change this stuff. See #198.